product_category_company/models/product_category.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+    _check_company_auto = True
+
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        index=True,
+    )
+    parent_id = fields.Many2one(check_company=True)
+
+    @api.constrains("parent_id", "company_id")
+    def check_company_restriction(self):
```

Comment:
Is this entire function even needed? `check_company=True` is set on `parent_id`, which should do all the work for us.

Reply:
ping @Kev-Roche @sebastienbeau could you have a look here, Thanks.

Is suite tests is OK without this method ?

Reply:
I will test this in a moment. I noticed this while busy with something else :)

Reply:
Removed the function in [67fb21d](https://github.com/OCA/multi-company/pull/470/commits/67fb21d5cdc65a65c74d4e0f554dd40e1f2a6e2b). Tests still run, although as a side effect, the error messages may be different.

![image](https://user-images.githubusercontent.com/12065945/232042477-a13c11e2-2a22-4900-abcd-7cd3a08f7040.png)

Reply:
No matter if message is different. Thanks